ELECTRON_RUN_AS_NODE is removed too late before bootstrap-amd.js redefines 'fs'

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- 🕮 Read our guide about submitting issues: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions -->
<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Insiders build to see if your issue has already been fixed: https://code.visualstudio.com/insiders/ -->
<!-- 💡 Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information. -->
<!-- 🔧 Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: Yes/No

<!-- 🪓 If you answered No above, use 'Help: Start Extension Bisect' from Command Palette to try to identify the cause. -->
<!-- 📣 Issues caused by an extension need to be reported directly to the extension publisher. The 'Help > Report Issue' dialog can assist with this. -->
- VS Code Version: stable
- OS Version:

Steps to Reproduce:

Regression from https://github.com/microsoft/vscode/pull/221154 

This broke my extension as I invoke the server-cli, now I get
```
2024-08-08 13:26:09.718 [error] Failed to initialize remote extensions: Loading "original-fs" failed
Error: Cannot find module 'original-fs'
Require stack:
- /home/gitpod/.vscode-server/bin/b1c0a14de1414fcdaa400695b4db1c0799bc3124/out/server-cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
```

cc @bpasero could this be fixed in recovery please

--- a/src/server-cli.js
+++ b/src/server-cli.js
@@ -6,6 +6,10 @@
 // @ts-check
 'use strict';
 
+// Keep bootstrap-amd.js from redefining 'fs'.
+// TODO@esm this needs to be revisited in ESM
+delete process.env['ELECTRON_RUN_AS_NODE'];
+
 // ESM-comment-begin
 const path = require('path');
 const bootstrapNode = require('./bootstrap-node');
@@ -25,9 +29,6 @@ const product = require('./bootstrap-meta').product;
 // ESM-uncomment-end
 
 async function start() {
-
-	// Keep bootstrap-amd.js from redefining 'fs'.
-	delete process.env['ELECTRON_RUN_AS_NODE'];
 
 	// NLS
 	const nlsConfiguration = await resolveNLSConfiguration({ userLocale: 'en', osLocale: 'en', commit: product.commit, userDataPath: '', nlsMetadataPath: __dirname });

--- a/src/server-main.js
+++ b/src/server-main.js
@@ -11,6 +11,10 @@
  * @import { IServerAPI } from './vs/server/node/remoteExtensionHostAgentServer'
  */
 
+// Keep bootstrap-amd.js from redefining 'fs'.
+// TODO@esm this needs to be revisited in ESM
+delete process.env['ELECTRON_RUN_AS_NODE'];
+
 // ESM-comment-begin
 const path = require('path');
 const http = require('http');
@@ -279,7 +283,6 @@ async function findFreePort(host, start, end) {
  */
 function loadCode(nlsConfiguration) {
 	return new Promise((resolve, reject) => {
-		delete process.env['ELECTRON_RUN_AS_NODE']; // Keep bootstrap-amd.js from redefining 'fs'.
 
 		/** @type {INLSConfiguration} */
 		process.env['VSCODE_NLS_CONFIG'] = JSON.stringify(nlsConfiguration); // required for `bootstrap-amd` to pick up NLS messages

